File: Cpp/242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> ms;
        for(auto el : s){
            ms[el]++;
        }
        map<char, int> mt;
        for(auto el : t){
            mt[el]++;
        }
        if(ms.size() != mt.size())return false;
        for(int i = 0; i < ms.size(); i++){
            if(ms[i] != mt[i])return false;
        }
        return true;
    }
};

File: Python/242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        ms = {}
        mt = {}
        for i, ch in enumerate(s):
            if ch not in ms:
                ms[ch] = 1
            else:
                ms[ch] = ms[ch] + 1
            ch = t[i]
            if ch not in mt:
                mt[ch] = 1
            else:
                mt[ch] = mt[ch] + 1
        for k, v in ms.items():
            if k in ms and k in mt:
                if ms[k] != mt[k]:
                    return False
            else:
                return False
        return True
